File: Programmers/Level_1/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    var temp = "";

    for (var i = 0; i < s.length; i++) {
        if (temp.lastIndexOf(s[i]) === -1) {
            answer.push(-1);
        } else {
            answer.push(i - temp.lastIndexOf(s[i]));
        }

        temp += s[i];
    }

    return answer;
}

File: Programmers/Level_1/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s)
{
    vector<int> answer;
    string temp = "";

    for (int i = 0; i < s.size(); i++)
    {
        bool flag = false;
        for (int j = i; j >= 0; j--)
        {
            if (temp[j - 1] == s[i])
            {
                flag = true;
                answer.push_back(i - j + 1);
                break;
            }
        }
        if (!flag)
        {
            answer.push_back(-1);
        }
        temp += s[i];
    }

    return answer;
}